### PR TITLE
feat: cli validate command

### PR DESCRIPTION
**Pull request by rephus**
Closes: #5229

### Description:

image

Screenshot from 2023-05-26 10-49-55

New version: 

image

### CODE REVIEW

File: packages/cli/src/handlers/dbt/validate.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    if (validation.length === 0) {
+        spinner?.succeed(`  Validation finished without errors`);
+    } else {
+        spinner?.fail(`  Validation finished with errors\n`);
+
+        validation.forEach((v) => {
+            console.error(
+                `- ${styles.error(v.name)}: ${styles.warning(v.error)}`,
+            );
+        });
```
Comment (ZeRego) [inline packages/cli/src/handlers/dbt/validate.ts:107]: We could use a library like this one to create a table/columns https://www.npmjs.com/package/columnify

File: packages/backend/src/controllers/validationController.ts [hunk 2]
```diff
@@ -34,12 +35,17 @@ export class ValidationController extends Controller {
     async post(
         @Path() projectUuid: string,
         @Request() req: express.Request,
+        @Query() context?: ValidateProjectPayload['context'],
```
Comment (ZeRego) [inline packages/backend/src/controllers/validationController.ts:38]: We can use the existing header information to detect if a request comes from the CLI, APP, other. This header is available in all requests.

Example in the projectRouter
```getRequestMethod(req.header(LightdashRequestMethodHeader)),```

With the controller, it should be possible to have something like:
```
@Header(LightdashRequestMethodHeader) context?: string;
...

jobId: await validationService.validate(
                    req.user!,
                    projectUuid,
                    context: getRequestMethod(context),
                ),
```

Comment (rephus) [reply packages/backend/src/controllers/validationController.ts:38]: good idea ! I'll do it this way

**Review by ZeRego — APPROVED** [review]

**Comment by rephus** [comment]
@TuringLovesDeathMetal chart/dashboard urls don't look great on the CLI output 
image

**Comment by TuringLovesDeathMetal** [comment]
oof, ya, let's keep the links out of there

**Comment by TuringLovesDeathMetal** [comment]
@rephus related to this comment:

> we don't return success on validation, only errors, I can still filter by chart/dashboard/Table and show you a summary before the table if you want

Yes! That would be really helpful :) Once we implement that, I'm happy with it!

**Pull request merged**